info-compiler/src/main/java/com/google/sps/infocompiler/InfoCompiler.java

```diff
@@ -270,28 +274,39 @@ void storeElectionContestInDatabase(String electionQueryId, JsonObject contest)
     datastore.update(electionEntity);
   }
 
-  private boolean checkIncumbency(String candidateName, String position, String division) {
+  private boolean checkIncumbency(Map<String, String> map, String position, String name) {
+    for (Map.Entry<String, String> e : map.entrySet()) {
+      if (e.getKey() == position && e.getValue() == name) return true;
+    }
+    return false;
+  }
+  private Map<String, String> getincumbents(String division) {
+    Map<String, String> mapOfIncumbents = new HashMap<>();  
     try {
       String queryUrl = String.format("%s&ocdDivisionId=%s", REPRENTATIVE_QUERY_URL, division);
       JsonObject representatives = queryCivicInformation(queryUrl);
       JsonArray offices = representatives.getAsJsonArray("offices");
-      int i = 0;
-      for (JsonElement eachOffice : offices){
-        JsonObject office = (JsonObject) eachOffice;
-        String officeName = office.get("name").getAsString();
-        if (position.equals(officeName)){
-          break;
+      JsonObject office;
+      String officeName;
+      JsonArray officialIndicesArray;
+      JsonArray officials;
+      for (JsonElement eachOffice : offices) {
+        office = (JsonObject) eachOffice;
+        officeName = office.get("name").getAsString();
+        officialIndicesArray =  office.getAsJsonArray("officialIndices");
+        officials = office.getAsJsonArray("officials");
```

Comment:
Why not just do a for each loop through "officials" instead of also getting the indices?  What do the indices add?

Reply:
The indices indicate which office each individual belongs to. If we don't  consider the indices given in the offices array then we will loop through officials and assign one official per office. Unless there is always one official per office this wont work.

Reply:
Oh gotcha, just read the API documentation again (didn't realize it was so weird haha).  I think though that `officials` should be populated before the for loop since it's at the same level as offices (i.e. `JsonArray officials = representatives.getAsJsonArray("officials")`).